tensorflow/core/nccl/nccl_manager.cc

```diff
@@ -410,7 +410,13 @@ void NcclManager::AddParticipant(std::unique_ptr<Participant> participant,
                                  CollectiveType collective_type,
                                  ncclRedOp_t reduction_op) {
   Collective* to_run = nullptr;
-  const DataType data_type = participant->input->dtype();
+  DataType data_type_val;
+  if (participant->input != nullptr) {
+    data_type_val = participant->input->dtype();
+  } else {
+    data_type_val = participant->output->dtype();
+  }
+  const DataType data_type = data_type_val;
```

Comment:
Let's get rid of this extra copy and just use the non-const DataType below.

Reply:
Yes. Fixed. Please have a look.